### PR TITLE
New Resource: `azurerm_netapp_volume_bucket`, `azurerm_netapp_volume_bucket_credentials` and `azurerm_netapp_volume_bucket` data source - Object REST API support

### PR DESCRIPTION
**Pull request by paulomarquesc**
## Community Note
* Please vote on this PR by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Description

This PR introduces support for the Azure NetApp Files **Object REST API** (preview) feature, exposed in ARM as the `Buckets` child resource of a Volume. Buckets provide S3-compatible object access to the data inside an existing Azure NetApp Files volume so that external services (Azure Databricks, Microsoft Fabric, OneLake, Azure AI Search, generic S3 clients, etc.) can consume volume data over HTTPS.

Public reference: [Configure object REST API for Azure NetApp Files (preview)](https://learn.microsoft.com/en-us/azure/azure-netapp-files/object-rest-api-access-configure)

This PR also bumps the NetApp SDK from `2025-06-01` to `2026-01-01`, which is the API version that introduces the `buckets` package required by this feature. All existing NetApp resources/data sources have been updated accordingly.

### What's added

**SDK**
* `netapp` - bumps the API version from `2025-06-01` to `2026-01-01`.

**New resources**
* **New Resource**: `azurerm_netapp_volume_bucket` - manages an S3-compatible bucket on top of a NetApp volume. Supports both inline `server.certificate_pem` upload and Azure Key Vault-sourced certificate (`key_vault {}`).
* **New Resource**: `azurerm_netapp_volume_bucket_credentials` - generates the access key / secret key pair used to authenticate against a bucket. Supports inline keys (returned as sensitive attributes) or Azure Key Vault-stored keys via `store_in_key_vault = true`.

**New data source**
* **New Data Source**: `azurerm_netapp_volume_bucket` - reads bucket configuration and runtime metadata (status, server IP, certificate CN/expiry).

**Examples**
* `examples/netapp/volume_bucket` - inline certificate + inline credentials.
* `examples/netapp/volume_bucket_akv` - Key Vault-sourced certificate, Key Vault-stored credentials, with the parent NetApp account configured with a system-assigned managed identity. Recommended for production use.

### Notes for reviewers
* The Object REST API preview feature must be registered on the subscription via `Microsoft.NetApp / ANFEnableObjectRESTAPI` before the API will accept bucket creates. This is documented in both example READMEs and in the resource markdown.
* The first bucket created on a set of volumes that share a backing IP must always supply both `server.fqdn` and either `server.certificate_pem` or a `key_vault {}` block. Subsequent buckets on the same backing IP can omit them. Surfaced as a note in the resource doc.
* When `key_vault {}` is used, the parent NetApp account needs a system-assigned managed identity with the documented Key Vault certificate / secret permissions. The AKV example wires this up end-to-end.
* `Update()` is intentionally not implemented on `azurerm_netapp_volume_bucket_credentials` because every Optional/Required field is `ForceNew` (rotation = replacement; the API has no separate revoke operation). The `Delete()` is a no-op and documented as such.

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work.

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevant documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.

## Testing

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass.

### Summary of test results

```
=== RUN   TestAccNetAppVolumeBucketCredentials_basic
=== PAUSE TestAccNetAppVolumeBucketCredentials_basic
=== RUN   TestAccNetAppVolumeBucket_basic
=== PAUSE TestAccNetAppVolumeBucket_basic
=== RUN   TestAccNetAppVolumeBucket_complete
=== PAUSE TestAccNetAppVolumeBucket_complete
=== RUN   TestAccNetAppVolumeBucket_update
=== PAUSE TestAccNetAppVolumeBucket_update
=== RUN   TestAccNetAppVolumeBucket_withKeyVault
=== PAUSE TestAccNetAppVolumeBucket_withKeyVault
=== RUN   TestAccNetAppVolumeBucket_requiresImport
=== PAUSE TestAccNetAppVolumeBucket_requiresImport
=== CONT  TestAccNetAppVolumeBucketCredentials_basic
--- PASS: TestAccNetAppVolumeBucketCredentials_basic (939.49s)
=== CONT  TestAccNetAppVolumeBucket_update
--- PASS: TestAccNetAppVolumeBucket_update (1006.56s)
=== CONT  TestAccNetAppVolumeBucket_requiresImport
--- PASS: TestAccNetAppVolumeBucket_requiresImport (875.31s)
=== CONT  TestAccNetAppVolumeBucket_withKeyVault
--- PASS: TestAccNetAppVolumeBucket_withKeyVault (1389.39s)
=== CONT  TestAccNetAppVolumeBucket_complete
=== CONT  TestAccNetAppVolumeBucket_basic
--- PASS: TestAccNetAppVolumeBucket_complete (908.01s)
--- PASS: TestAccNetAppVolumeBucket_basic (901.08s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/netapp        6019.865s
```

Total tests run: 6 | Passed: 6 | Failed: 0 | Pass rate: 100%

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-merging.md).

* `dependencies`: `netapp` - update API version to `2026-01-01`
* **New Resource**: `azurerm_netapp_volume_bucket`
* **New Resource**: `azurerm_netapp_volume_bucket_credentials`
* **New Data Source**: `azurerm_netapp_volume_bucket`

This is a (please select all that apply):

- [ ] Bug Fix
- [x] New Feature (ie adding a service, resource, or data source)
- [ ] Enhancement
- [ ] Breaking Change

## Related Issue(s)

N/A

## AI Assistance Disclosure

- [x] AI Assisted - This contribution was made by, or with the assistance of, AI/LLMs

AI was used to scaffold the new typed resources / data source from the existing typed `sdk.Resource` patterns in the netapp service, draft documentation and examples, and assist with iterative refinement of CustomizeDiff, schema layout, and acceptance test scaffolding. All changes were reviewed, refined, and validated by the author against the actual Azure NetApp Files Object REST API behaviour through real acceptance test runs.

## Rollback Plan

If a change needs to be reverted, we will publish an updated version of the provider.

## Changes to Security Controls

This PR introduces the ability to:
* Generate S3 access keys / secret keys for buckets, optionally persisted to Azure Key Vault (recommended) or returned inline as sensitive Terraform state attributes.
* Source the bucket server TLS certificate from Azure Key Vault, with the parent NetApp account using a system-assigned managed identity to read it. The example demonstrates the documented set of Key Vault permissions required.

The Object REST API feature itself is gated by the subscription-level preview registration `Microsoft.NetApp / ANFEnableObjectRESTAPI`.

### CODE REVIEW

**Comment by paulomarquesc** [comment]
## List resource update

Pushed [577c64b](https://github.com/hashicorp/terraform-provider-azurerm/pull/32319/commits/577c64befc) which adds:

- Resource Identity (`sdk.ResourceWithIdentity`) on `azurerm_netapp_volume_bucket`
- Shared `flatten` helper extracted from `Read`
- New list resource `azurerm_netapp_volume_bucket` scoped by `volume_id` (uses `Buckets_List` from the SDK)
- `TestAccNetAppVolumeBucket_list` acceptance test
- `website/docs/list-resources/netapp_volume_bucket.html.markdown`

### Request: `list-not-supported` for `azurerm_netapp_volume_bucket_credentials`

Could a maintainer please apply the `list-not-supported` label to this PR for the `azurerm_netapp_volume_bucket_credentials` resource?

This resource is a synthetic Terraform wrapper around the synchronous `Buckets_GenerateCredentials` POST operation (or its async `Buckets_GenerateAkvCredentials` LRO counterpart). The Azure NetApp Files Object REST API does not expose a corresponding `List` operation for bucket credentials:

- Each invocation of the credentials API generates a new key pair and immediately invalidates any prior key pair for the same bucket.
- Generated credentials are never re-readable once returned (or, in the AKV case, once written to Key Vault).
- There is no enumeration endpoint and no equivalent of `Buckets_List` for credential pairs.

The resource is also fully `ForceNew` (every Required/Optional field rotates the credential pair) and has no `Update()` method by design - implementing `List` is not feasible for this resource type.

File: internal/services/netapp/netapp_volume_bucket_credentials_resource.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/netapp/2026-01-01/buckets"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	netAppModels "github.com/hashicorp/terraform-provider-azurerm/internal/services/netapp/models"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type NetAppVolumeBucketCredentialsResource struct{}
```
Comment (mbfrahry) [inline internal/services/netapp/netapp_volume_bucket_credentials_resource.go]: I think this resource seems better suited as a Terraform Action since it's generating credentials as a one off operation. We lose the ability to  generate and store credentials into state but from the docs, that's not really production ready so maybe we can nix that path in favor of only support generating credentials for keyvault?

Regardless,  I suggest removing `store_in_key_vault` from this resource and putting that into an Action

File: internal/services/netapp/netapp_volume_bucket_data_source.go [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: netAppValidate.PoolName,
+		},
+
+		"volume_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: netAppValidate.VolumeName,
+		},
```
Comment (mbfrahry) [inline internal/services/netapp/netapp_volume_bucket_data_source.go]: If a bucket is tied to a volume, could we consolidate all this down into `netapp_volume_id`?

File: internal/services/netapp/netapp_volume_bucket_resource.go [hunk 3]
```diff
@@ -0,0 +1,570 @@
...
+			ValidateFunc: volumes.ValidateVolumeID,
+		},
+
+		"file_system_user": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"nfs_user": {
```
Comment (mbfrahry) [inline internal/services/netapp/netapp_volume_bucket_resource.go]: Could we move `nfs_user` and `cifs_user` up a block to look like `file_system_nfs_user` and file_system_cifs_user`? At that point we can do ExactlyOneOf on that block and make this simpler

File: internal/services/netapp/netapp_volume_bucket_resource.go [hunk 4]
```diff
@@ -0,0 +1,570 @@
...
+						Optional:      true,
+						Sensitive:     true,
+						ValidateFunc:  validation.StringIsBase64,
+						ConflictsWith: []string{"key_vault"},
+					},
+					"on_certificate_conflict_action": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						Default:      string(buckets.OnCertificateConflictActionFail),
+						ValidateFunc: validation.StringInSlice(buckets.PossibleValuesForOnCertificateConflictAction(), false),
```
Comment (mbfrahry) [inline internal/services/netapp/netapp_volume_bucket_resource.go]: Are any of these properties required in the `server` block? I'm reading it as an Optional block with all Optional properties which is something we try to avoid. 

The basic test is setting this block which is also something we avoid as the basic test should only set properties that are the bare minimum to get the resource stood up. As written, someone could pass in:
```
  server {}
```

File: internal/services/netapp/netapp_volume_bucket_resource.go [hunk 5]
```diff
@@ -0,0 +1,570 @@
...
+		},
+
+		"status": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+	}
+}
+
+func (r NetAppVolumeBucketResource) CustomizeDiff() sdk.ResourceFunc {
```
Comment (mbfrahry) [inline internal/services/netapp/netapp_volume_bucket_resource.go]: We won't need this CustomizeDiff if we pull out the users into their own top level properties

File: internal/services/netapp/netapp_volume_bucket_resource_list.go [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+		return
+	}
+
+	r := NetAppVolumeBucketResource{}
+	stream.Results = func(push func(list.ListResult) bool) {
+		for _, item := range resp.Items {
+			result := request.NewListResult(ctx)
+			result.DisplayName = pointer.From(item.Name)
+
+			id, err := buckets.ParseBucketIDInsensitively(pointer.From(item.Id))
```
Comment (mbfrahry) [inline internal/services/netapp/netapp_volume_bucket_resource_list.go:80]: Do we need to parse the id insensitively?

File: internal/services/netapp/netapp_volume_bucket_resource_test.go [hunk 7]
```diff
@@ -0,0 +1,439 @@
...
+
+func (NetAppVolumeBucketResource) complete(data acceptance.TestData) string {
+	template := NetAppVolumeBucketResource{}.template(data)
+	return fmt.Sprintf(`
+%[1]s
+
+resource "azurerm_netapp_volume_bucket" "test" {
+  name        = "acctest-bucket-%[2]d"
+  volume_id   = azurerm_netapp_volume.test.id
+  path        = "/"
```
Comment (mbfrahry) [inline internal/services/netapp/netapp_volume_bucket_resource_test.go]: This is redundant as the default of path is `/`

**Review by mbfrahry — CHANGES_REQUESTED** [review]
Hey @paulomarquesc, I've left some design questions here and a potential rework of the credentials resource into an action. Also, this PR should be split into multiple prs to make it easier to review. The API bump should be it's own PR in case that bump causes any issues in other resources.  And then the bucket/credentials pr into their own prs.

File: internal/services/netapp/netapp_volume_bucket_resource.go (hunk 4 above)
Comment (paulomarquesc) [reply internal/services/netapp/netapp_volume_bucket_resource.go]: Yes, they are all optional because there is a big catch that we can't work around, for the first volume within an SVM (an internal ONTAP construct, not exposed in Azure) they are required but there is no way to know if it is the first one or not, if that was possible, we could use CustomDiff logic, so to make it work, we actually need this as-is and the basic test must provide it as it is the first volume, so I won't be able to modify this.

**Comment by paulomarquesc** [comment]
> Member

Thanks Matthew, for next time, I will create separate PRs.

File: internal/services/netapp/netapp_volume_bucket_resource_list.go (hunk 6 above)
Comment (paulomarquesc) [reply internal/services/netapp/netapp_volume_bucket_resource_list.go]: Is there a downside to using this?

Comment (paulomarquesc) [reply internal/services/netapp/netapp_volume_bucket_resource_list.go:80]: This pattern is used extensively in other services:

image

What is the downside of using it here? It found 4405 occurrences of this pattern over ~2K files.

**Comment by paulomarquesc** [comment]
@mbfrahry , thanks for the review and great insights, I've pushed all changes, but I left a few comments in response to the review.

File: internal/services/netapp/netapp_volume_bucket_credentials_resource.go (hunk 1 above)
Comment (paulomarquesc) [reply internal/services/netapp/netapp_volume_bucket_credentials_resource.go]: Done.

File: internal/services/netapp/netapp_volume_bucket_data_source.go (hunk 2 above)
Comment (paulomarquesc) [reply internal/services/netapp/netapp_volume_bucket_data_source.go]: Done.

File: internal/services/netapp/netapp_volume_bucket_resource.go (hunk 3 above)
Comment (paulomarquesc) [reply internal/services/netapp/netapp_volume_bucket_resource.go]: done

File: internal/services/netapp/netapp_volume_bucket_resource.go (hunk 5 above)
Comment (paulomarquesc) [reply internal/services/netapp/netapp_volume_bucket_resource.go]: Done.

File: internal/services/netapp/netapp_volume_bucket_resource_test.go (hunk 7 above)
Comment (paulomarquesc) [reply internal/services/netapp/netapp_volume_bucket_resource_test.go]: Done.

File: internal/services/netapp/netapp_volume_bucket_resource_list.go (hunk 6 above)
Comment (mbfrahry) [reply internal/services/netapp/netapp_volume_bucket_resource_list.go:80]: It can be an issue if a service expects a certain case. We try to match case where necessary in case we read something in insensitively and that gets passed to a service that is expecting case in a specific way. 

In some of the cases you're describing, Azure doesn't respect the case being sent to it which causes a whole host of other issues so we have to parse insensitively. 

I'm just asking here in case there is an issue with case sensitivity and we can mark down that reason why here

File: internal/services/netapp/netapp_volume_bucket_resource.go (hunk 4 above)
Comment (mbfrahry) [reply internal/services/netapp/netapp_volume_bucket_resource.go]: I'm not entirely sure I follow this. Is `server {}` a valid configuration?

If not, and at least one of these values is Required, we could do `AtLeastOneOf: []string{"server.0.fqdn"...}`  on all of these attributes.

**Review by mbfrahry — COMMENTED** [review]
Hey @paulomarquesc, thanks for making those changes! This is looking nearly there! Just a couple more comments to follow up on and some merge conflicts to fix up. Als can you also post a full netapp test suite run since this is bumping the api version

File: internal/services/netapp/netapp_volume_bucket_resource.go (hunk 4 above)
Comment (paulomarquesc) [reply internal/services/netapp/netapp_volume_bucket_resource.go]: All these are optional for the 2nd volume onwards, I know this is not great but that's how the underlying storage OS (ONTAP) behaves, so we really must do this way.

File: internal/services/netapp/netapp_volume_bucket_resource_list.go (hunk 6 above)
Comment (paulomarquesc) [reply internal/services/netapp/netapp_volume_bucket_resource_list.go:80]: There is no particular reason, so I thought it should do no harm leaving like that, my search shows thousands of other code paths using it, but if you feel strong about it, I can fallback to regular parse.

Comment (mbfrahry) [reply internal/services/netapp/netapp_volume_bucket_resource_list.go:80]: I would prefer regular parse unless it's necessary to have it parsed insensitively

**Comment by paulomarquesc** [comment]
Tests results and summary of the failures.

Test results:

```
--- PASS: TestAccNetAppAccountResource (592.02s)
    --- PASS: TestAccNetAppAccountResource/Resource (0.00s)
        --- PASS: TestAccNetAppAccountResource/Resource/requiresImport (148.47s)
    --- PASS: TestAccNetAppAccountResource/Resource#01 (0.00s)
        --- PASS: TestAccNetAppAccountResource/Resource#01/complete (135.13s)
    --- PASS: TestAccNetAppAccountResource/Resource#02 (0.00s)
        --- PASS: TestAccNetAppAccountResource/Resource#02/update (174.14s)
    --- PASS: TestAccNetAppAccountResource/Resource#03 (0.00s)
        --- PASS: TestAccNetAppAccountResource/Resource#03/basic (134.28s)
--- PASS: TestAccNetAppVolumeBucket_list (782.72s)
--- PASS: TestAccNetAppAccountEncryptionDataSource_basic (369.37s)
--- FAIL: TestAccNetAppVolumeGroupOracle_crossRegionReplication (1002.13s)
--- PASS: TestAccNetAppVolume_advancedRansomwareProtectionUpdate (886.15s)
--- PASS: TestAccNetAppVolume_advancedRansomwareProtection (728.37s)
--- PASS: TestAccNetAppVolume_protocolConversion (1047.39s)
--- PASS: TestAccNetAppVolume_coolAccess (1332.06s)
--- PASS: TestAccNetAppVolume_serviceLevelUpdate (978.33s)
--- PASS: TestAccNetAppVolume_volEncryptionCmkSystemAssigned (1091.73s)
--- PASS: TestAccNetAppVolume_volEncryptionCmkUserAssigned (1070.23s)
--- PASS: TestAccNetAppVolume_updateExportPolicyRule (889.67s)
--- PASS: TestAccNetAppVolume_update (906.72s)
--- PASS: TestAccNetAppVolume_largeVolume (766.93s)
--- PASS: TestAccNetAppVolume_complete (741.85s)
--- PASS: TestAccNetAppVolumeQuotaRule_defaultGroupQuotaType (930.81s)
--- PASS: TestAccNetAppVolumeQuotaRule_individualUserQuotaType (949.63s)
--- PASS: TestAccNetAppVolumeQuotaRule_individualGroupQuotaType (948.26s)
--- PASS: TestAccNetAppVolumeQuotaRuleDataSource_basic (911.80s)
--- PASS: TestAccNetAppVolumeGroupSAPHana_crossZoneReplicationAZ (2394.54s)
--- FAIL: TestAccNetAppVolumeGroupSAPHana_crossRegionReplicationAZ (1057.41s)
--- PASS: TestAccNetAppVolumeGroupSAPHana_crossRegionReplication (2594.28s)
--- PASS: TestAccNetAppVolumeGroupSAPHana_protocolConversion (1770.98s)
--- PASS: TestAccNetAppVolumeGroupSAPHana_volumeUpdates (1531.57s)
--- PASS: TestAccNetAppVolumeGroupSAPHana_snapshotPolicyUpdate (1849.81s)
--- PASS: TestAccNetAppVolumeGroupSAPHana_snapshotPolicy (1350.73s)
--- PASS: TestAccNetAppVolumeGroupSAPHana_backupVolumeSpecsNfsv3 (1491.90s)
--- PASS: TestAccNetAppVolumeGroupSAPHana_volCustomerManagedKeyEncryption (1412.53s)
--- PASS: TestAccNetAppVolumeGroupSAPHana_basicAvailabilityZone (1115.91s)
--- PASS: TestAccNetAppVolumeGroupSAPHana_basic (1208.15s)
--- PASS: TestAccNetAppVolumeGroupSAPHanaDataSource_basic (1111.15s)
--- PASS: TestAccNetAppVolumeGroupOracle_crossZoneReplicationAZ (2299.09s)
--- FAIL: TestAccNetAppVolumeGroupOracle_crossRegionReplicationAZ (1022.86s)
--- PASS: TestAccNetAppVolume_snapshotPolicy (805.05s)
--- PASS: TestAccNetAppVolume_requiresImport (749.26s)
--- PASS: TestAccNetAppVolume_nfsv3SnapshotDirectoryVisibleTrue (749.85s)
--- PASS: TestAccNetAppVolume_nfsv3SnapshotDirectoryVisibleFalse (741.37s)
--- PASS: TestAccNetAppVolume_nfsv3FromSnapshot (1019.85s)
--- FAIL: TestAccNetAppVolume_crossZoneRegionReplication (3038.54s)
--- PASS: TestAccNetAppVolume_crossZoneReplication (1668.81s)
--- FAIL: TestAccNetAppVolume_crossRegionReplication (3020.11s)
--- PASS: TestAccNetAppVolume_snapshotPolicyUpdate (989.25s)
--- PASS: TestAccNetAppSnapshotPolicy_weeklySchedule (301.26s)
--- PASS: TestAccNetAppVolumeGroupOracle_volCustomerManagedKeyEncryption (1234.05s)
--- PASS: TestAccNetAppVolumeGroupOracle_protocolConversion (1580.52s)
--- PASS: TestAccNetAppVolumeGroupOracle_volumeUpdates (1314.40s)
--- PASS: TestAccNetAppVolumeGroupOracle_snapshotPolicyUpdate (1648.95s)
--- PASS: TestAccNetAppVolumeGroupOracle_snapshotPolicy (1253.41s)
--- PASS: TestAccNetAppVolumeGroupOracle_nfsv3 (1064.56s)
--- PASS: TestAccNetAppVolumeGroupOracle_basicProximityPlacementGroup (1151.20s)
--- PASS: TestAccNetAppVolumeGroupOracle_basicAvailabilityZone (1050.62s)
--- PASS: TestAccNetAppVolumeGroupOracleDataSource_basic (1012.92s)
--- PASS: TestAccNetAppVolumeBucket_serverBlockVariations (1109.38s)
--- PASS: TestAccNetAppVolumeBucket_requiresImport (823.43s)
--- PASS: TestAccNetAppVolumeBucket_withKeyVault (1353.75s)
--- PASS: TestAccNetAppVolumeBucket_update (976.44s)
--- PASS: TestAccNetAppVolumeBucket_complete (866.46s)
--- PASS: TestAccNetAppVolumeQuotaRule_defaultUserQuotaType (936.26s)
--- PASS: TestAccNetAppVolume_backupPolicyUpdate (1734.02s)
```

Overall result: FAIL (run time: 72000.095s ≈ 20h, hit timeout)

Result table

 **Overall result:** `FAIL` (run time: 72000.095s ≈ 20h, hit timeout)
 
 ## Result table
 
 | Outcome | Count |
 |---|---:|
 | PASS  | 54 |
 | FAIL  |  5 |
 | SKIP  |  0 |
 | **Total** | **59** |
 
 ## Failures
 
 | # | Test | Duration | Root cause |
 |---|---|---:|---|
 | 1 | `TestAccNetAppVolumeGroupOracle_crossRegionReplication`      | 1002.13s | AZ mapping not available (zone 1) in `australiasoutheast` for sub `66bc9830-…` |
 | 2 | `TestAccNetAppVolumeGroupSAPHana_crossRegionReplicationAZ`   | 1057.41s | AZ mapping not available (zone 2) in `australiasoutheast` for sub `66bc9830-…` |
 | 3 | `TestAccNetAppVolumeGroupOracle_crossRegionReplicationAZ`    | 1022.86s | AZ mapping not available (zone 2) in `australiasoutheast` for sub `66bc9830-…` |
 | 4 | `TestAccNetAppVolume_crossZoneRegionReplication`             | 3038.54s | `eastus2` create-volume conflict: "Another volume creation is in progress"; post-test destroy of capacity pool also hit `context deadline exceeded` |
 | 5 | `TestAccNetAppVolume_crossRegionReplication`                 | 3020.11s | `eastus2` create-volume conflict: "Another volume creation is in progress"; post-test destroy of capacity pool also hit `context deadline exceeded` |

**Comment by paulomarquesc** [comment]
New tests with the added extra resources:

```
pmarques [ ~ ]$ tail -f ~/tests.txt | grep -e "PASS:" -e "FAIL:"
--- PASS: TestAccNetAppVolumeBucket_list (808.06s)
--- PASS: TestAccNetAppVolumeBucket_complete (1612.38s)
--- PASS: TestAccNetAppVolumeBucketWithServer_basic (1662.00s)
--- PASS: TestAccNetAppVolumeBucket_update (1701.79s)
--- PASS: TestAccNetAppVolumeBucketCredentialsAction_basic (2246.23s)
--- PASS: TestAccNetAppVolumeBucketWithServer_withKeyVault (1344.22s)
--- PASS: TestAccNetAppVolumeBucketWithServer_requiresImport (1602.51s)
--- PASS: TestAccNetAppVolumeBucket_requiresImport (1572.61s)
--- PASS: TestAccNetAppVolumeBucket_basic (1360.28s)
--- PASS: TestAccNetAppVolumeBucketWithServer_update (1001.48s)
--- PASS: TestAccNetAppVolumeBucketWithServer_complete (906.97s)
```

File: internal/services/netapp/netapp_volume_bucket_helper.go [hunk 8]
```diff
@@ -0,0 +1,403 @@
...
+						Type:         pluginsdk.TypeInt,
+						Required:     true,
+						ValidateFunc: validation.IntAtLeast(0),
+					},
+				},
+			},
+			ExactlyOneOf: []string{"file_system_nfs_user", "file_system_cifs_user"},
+		},
+
+		"file_system_cifs_user": {
```
Comment (mbfrahry) [inline internal/services/netapp/netapp_volume_bucket_helper.go]: Will this only hold one value? If so, we can just make this a string. This also isn't being tested
```suggestion
		"file_system_cifs_username": {
```

File: internal/services/netapp/netapp_volume_bucket_helper.go [hunk 9]
```diff
@@ -0,0 +1,403 @@
...
+					"credentials_secret_name": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+				},
+			},
+		},
+
+		"path": {
```
Comment (mbfrahry) [inline internal/services/netapp/netapp_volume_bucket_helper.go:94]: Can we add a test for this property?

File: website/docs/r/netapp_volume_bucket_with_server.html.markdown [hunk 10]
```diff
@@ -0,0 +1,404 @@
...
+
+* `file_system_cifs_user` - (Optional) A `file_system_cifs_user` block as defined below. Exactly one of `file_system_nfs_user` or `file_system_cifs_user` must be specified.
+
+---
+
+* `path` - (Optional) The volume sub-path mounted inside the bucket. Defaults to `/`. Changing this forces a new resource to be created.
+
+* `permissions` - (Optional) The bucket permission level. Possible values are `ReadOnly` and `ReadWrite`. Defaults to `ReadOnly`.
+
+* `server` - (Required) A `server` block as defined below. Used to provide the bucket server FQDN and a directly uploaded PEM certificate. The certificate source (`server.0.certificate_pem`) is mutually exclusive with `key_vault`.
```
Comment (mbfrahry) [inline website/docs/r/netapp_volume_bucket_with_server.html.markdown]: Can we move this attribute up so we have all the Required properties together followed by the Optional properties

File: website/docs/r/netapp_volume_bucket_with_server.html.markdown [hunk 11]
```diff
@@ -0,0 +1,404 @@
...
+
+* `name` - (Required) The S3-compatible name of the bucket. Must be 3-63 characters long, DNS-compliant (lowercase letters, digits, hyphens or periods), must start and end with a letter or number and must not look like an IPv4 address. Changing this forces a new resource to be created.
+
+* `volume_id` - (Required) The ARM ID of the parent NetApp Volume the bucket attaches to. Changing this forces a new resource to be created.
+
+* `file_system_nfs_user` - (Optional) A `file_system_nfs_user` block as defined below. Exactly one of `file_system_nfs_user` or `file_system_cifs_user` must be specified.
+
+* `file_system_cifs_user` - (Optional) A `file_system_cifs_user` block as defined below. Exactly one of `file_system_nfs_user` or `file_system_cifs_user` must be specified.
+
+---
```
Comment (mbfrahry) [inline website/docs/r/netapp_volume_bucket_with_server.html.markdown]: I don't think we need to add a separator here
```suggestion
```

File: website/docs/r/netapp_volume_bucket.html.markdown [hunk 12]
```diff
@@ -0,0 +1,221 @@
...
+
+* `name` - (Required) The S3-compatible name of the bucket. Must be 3-63 characters long, DNS-compliant (lowercase letters, digits, hyphens or periods), must start and end with a letter or number and must not look like an IPv4 address. Changing this forces a new resource to be created.
+
+* `volume_id` - (Required) The ARM ID of the parent NetApp Volume the bucket attaches to. Changing this forces a new resource to be created.
+
+* `file_system_nfs_user` - (Optional) A `file_system_nfs_user` block as defined below. Exactly one of `file_system_nfs_user` or `file_system_cifs_user` must be specified.
+
+* `file_system_cifs_user` - (Optional) A `file_system_cifs_user` block as defined below. Exactly one of `file_system_nfs_user` or `file_system_cifs_user` must be specified.
+
+---
```
Comment (mbfrahry) [inline website/docs/r/netapp_volume_bucket.html.markdown]:
```suggestion
```

**Review by mbfrahry — CHANGES_REQUESTED** [review]
Hey @paulomarquesc, this is looking nearly there. Just a couple callouts and one piece that isn't being tested.

File: internal/services/netapp/netapp_volume_bucket_helper.go (hunk 8 above)
Comment (paulomarquesc) [reply internal/services/netapp/netapp_volume_bucket_helper.go]: Flattened to string.

File: internal/services/netapp/netapp_volume_bucket_helper.go (hunk 9 above)
Comment (paulomarquesc) [reply internal/services/netapp/netapp_volume_bucket_helper.go:94]: The problem of adding an acceptance test for this is that the folder must already exist inside the volume so we will need to 1) provision a VM, 2) mount the volume inside the VM, 3) create the folder, then run the acceptance test.

My take, this would be extremely overkill just to test a property and would avoid this, too many moving parts that are prone to all sorts of issues, so value in adding it.

File: website/docs/r/netapp_volume_bucket_with_server.html.markdown (hunk 10 above)
Comment (paulomarquesc) [reply website/docs/r/netapp_volume_bucket_with_server.html.markdown]: Resolved

File: website/docs/r/netapp_volume_bucket.html.markdown (hunk 12 above)
Comment (paulomarquesc) [reply website/docs/r/netapp_volume_bucket.html.markdown]: Resolved

File: website/docs/r/netapp_volume_bucket_with_server.html.markdown (hunk 11 above)
Comment (paulomarquesc) [reply website/docs/r/netapp_volume_bucket_with_server.html.markdown]: Resolved

File: internal/services/netapp/netapp_volume_bucket_helper.go (hunk 9 above)
Comment (paulomarquesc) [reply internal/services/netapp/netapp_volume_bucket_helper.go:94]: This is what we get if trying to create a bucket from a folder that does not exist:

```
The bucket 'acctest-bucket-260625042158513237' was not created
        successfully as the path '/data' does not exist in the underlying volume
        'acctest-NetAppAccount-260625042158513237/acctest-NetAppPool-260625042158513237/acctest-NetAppVolume-260625042158513237'.
        Please delete the failed bucket and create a new one with a path that exists
        on the volume, or leave the bucket path empty for full volume access.
```

**Review by mbfrahry — APPROVED** [review]
LGTM!
